docs/FeatureManagement/FeatureFlagv2.0.0.schema.json

```diff
@@ -0,0 +1,354 @@
+                  "type": "string",
+                  "title": "Client Filter Name",
+                  "description": "The name used to refer to and require a client filter.",
+                  "default": "",
+                  "examples": [
+                    "Percentage",
+                    "TimeWindow"
+                  ],
+                  "pattern": "^(.*)$"
+                },
+                "parameters": {
+                  "$id": "#/properties/conditions/properties/client_filters/items/properties/parameters",
+                  "type": "object",
+                  "title": "Client Filter Parameters",
+                  "description": "Custom parameters for a given client filter. A client filter can require any set of parameters of any type.",
+                  "required": [],
+                  "patternProperties": {
+                    "^.*$": {
+                      "anyOf": [
+                        {
+                          "type": "string"
+                        },
+                        {
+                          "type": "null"
+                        },
+                        {
+                          "type": "object"
+                        },
+                        {
+                          "type": "number"
+                        },
+                        {
+                          "type": "array"
+                        },
+                        {
+                          "type": "boolean"
+                        }
+                      ]
+                    }
+                  },
+                  "additionalProperties": false
+                }
+              }
+            }
+          }
+        }
+      },
+      "variants": {
+        "$id": "#/properties/variants",
+        "type": "array",
+        "title": "Variants Collection",
+        "description": "The list of variants defined for this feature. A variant represents a value of a feature flag that can be a string, a number, a boolean, or even a configuration object.",
+        "items": {
+            "$id": "#/properties/variants/items",
+            "type": "object",
+            "title": "Variant",
+            "required": [
+                "name"
+            ],
+            "properties":{
+                "name": {
+                    "$id": "#/properties/variants/items/properties/name",
+                    "type": "string",
+                    "title": "Variant Name",
+                    "description": "The name used to refer to a feature variant.",
+                    "default": "",
+                    "pattern": "^(.*)$"
+                },
+                "configuration_value": {
+                    "$id": "#/properties/variants/items/properties/configuration_value",
+                    "type": ["string", "null", "number", "object", "array", "boolean"],
+                    "title": "Variant Configuration Value",
+                    "description": "The inline value set for this feature variant."
```

Comment:
Do we need to specify the pattern for `configuration_value` here (like `parameters` of `client_filters`)?

Reply:
From what I understand, `parameters` always expects an object so it makes sense to have the `patternProperties`, but I didn't know if I need it for this case if it's sometimes an object, but `configuration_value` can be anything.